app/services/SessionService.scala

```diff
@@ -88,7 +88,14 @@ class SessionService @Inject() (appConfig: AppConfig, sessionRepository: Session
   private[services] def guidanceSession(session: Session)(implicit context: ExecutionContext): Future[RequestOutcome[GuidanceSession]] = {
     val processId = if (session.runMode.equals(Some(Debugging))) s"${session.processId}${processCacheRepository.DebugIdSuffix}" else session.processId
     processCacheRepository.get(processId, session.processVersion, session.timescalesVersion, session.ratesVersion).map{
-      case Right(cachedProcess) => Right(GuidanceSession(session, cachedProcess.process, cachedProcess.pageMap))
+      case Right(cachedProcess) =>
+        toPageHistory(Some(session.rawPageHistory), cachedProcess.pageMap, cachedProcess.process.meta.processCode) match {
+          case None =>
+            logger.error(s"ERROR:Conversion of RawPageHistory to PageHistory failed (None return)")
+            Left(RawPageHistoryError)
+          case pageHistoryOption =>
```

Comment:
Use Some(pageHistory) in the match then you can remove the exposed get() in call to GuidanceSession apply fn